### PR TITLE
adding deployment section and updating on-prem docs

### PR DESCRIPTION
**Pull request by klehmer**
## Description & motivation
This PR adds a "Deployment" section to the dbt Cloud docs with information on the various Deployment models currently supported.

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [ ] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [x ] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

## Checklist
If you added new pages (delete if not applicable):
- [x ] The page has been added to `website/sidebars.js`
- [ ] The new page has a unique filename

If you removed existing pages (delete if not applicable):
- [x ] The page has been removed from `website/sidebars.js`
- [ ] An entry has been added to `_redirects`

### CODE REVIEW

File: website/docs/docs/dbt-cloud/deployments/architecture.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+- **VPC**: In both hosted deployments, the dbt Cloud application infrastructure lives in an [AWS VPC](https://aws.amazon.com/vpc/) managed by Fishtown Analytics. One of the key differences between the Production and Single Tenant deployment is that the Single Tenant deployment provides a dedicated VPC for a single customer.
+- **EKS**: Hosted environments leverage [AWS Elastic Kubernetes Service](https://aws.amazon.com/eks/) to manage dbt Cloud application resources. EKS provides a high degree of reliability and scalability for the dbt Cloud application.
+- **CLB**: One or more [AWS Classic Load Balancers](https://aws.amazon.com/elasticloadbalancing/) living in a public subnet are leveraged in the hosted deployment environments to distribute incoming traffic across multple EC2 instances in the EKS cluster.
+- **EC2**: The hosted dbt Cloud deployments leverage a cluster of [AWS EC2](https://aws.amazon.com/ec2/) worker nodes to run the dbt Cloud application.
+- **EBS**: In order to store temporary in-memory application data, dbt Cloud leverages [AWS Elastic Block Store](https://aws.amazon.com/ebs/) mounted to the EC2 instances described above.
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/deployments/architecture.md]: is this right? we use ebs for in-memory application data?

File: website/docs/docs/dbt-cloud/deployments/architecture.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+- **EFS**: An [AWS Elastic File System](https://aws.amazon.com/efs/) is provisioned for hosted deployments to store and manage local files from the dbt Cloud developer environment.
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/deployments/architecture.md]:
```suggestion
- **EFS**: An [AWS Elastic File System](https://aws.amazon.com/efs/) is provisioned for hosted deployments to store and manage local files from the dbt Cloud IDE.
```

File: website/docs/docs/dbt-cloud/deployments/architecture.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+- **Object Storage**: dbt Cloud requires an S3-compatible Object Storage system for persisting run logs and artifacts.
+- **Storage Volumes**: dbt Cloud requires a Kubernetes storage provider capable of creating dynamic persistent volumes that can be mounted to multiple containers in R/W mode.
+
+### Data Warehouse Interaction
+
+dbt Cloud's primary role is as a data processor, not a data store. The dbt Cloud application enables users to dispatch SQL to the warehouse for transformation purposes. However, it is possible for users to dispatch SQL that returns customer data into the dbt Cloud application. This data is never persisted and will only exist in memory on the instance in question. In order to properly lock down customer data, it is critical that proper data warehouse permissioning is applied to prevent improper access or storage of sensitive data.
+
+### Hosted Network Architecture
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/deployments/architecture.md]: can we indent this section and `Customer Managed (General) Network Architecture` below and wrap them under a `### Deployment Architecture` heading?

File: website/docs/docs/dbt-cloud/on-premises/dependencies.md [hunk 4]
```diff
@@ -65,15 +65,15 @@ In order to perform basic maintenance and license checking, the following outbou
 
 - `api.replicated.com`: This endpoint services the license sync check and is used to pull down yaml files for app upgrades.
 
-### Optional External Dependencies
+## Optional External Dependencies
 
-#### Auth0
+### Auth0
```
Comment (drewbanin) [inline website/docs/docs/dbt-cloud/on-premises/dependencies.md]: fyi our new sso integrations do not use Auth0 - Cloud will negotiate SSO with the identity providers natively. I think we can either remove this, or just note that Auth0 is "legacy" and no longer required for new deployments?

**Review by drewbanin — COMMENTED** [review]
Just dropped some comments in here, but this broadly looks very good to me!!

Do we _want_ to document the airgapped model today? I'd be ok with keeping that page, but removing the sidebar link to it. I'm also ok with leaving the sidebar link in and just leaning on the big caveat at the top of the page :)

File: website/docs/docs/dbt-cloud/deployments/architecture.md (hunk 1 above)
Comment (klehmer) [reply website/docs/docs/dbt-cloud/deployments/architecture.md]: Good catch this was wrong. I removed "temporary in-memory"

File: website/docs/docs/dbt-cloud/on-premises/dependencies.md (hunk 4 above)
Comment (klehmer) [reply website/docs/docs/dbt-cloud/on-premises/dependencies.md]: I removed this section for now. I think we should add a section on Okta/GSuite (as there is config for on-prem in the admin console for those). I made a ticket for myself to add that in the next PR with the on-prem setup updates.

**Review by drewbanin — APPROVED** [review]

**Pull request merged**